packages/wonder-blocks-labeled-field/src/components/__tests__/labeled-field.test.tsx

```diff
@@ -215,11 +230,240 @@ describe("LabeledField", () => {
+                    description={description}
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
+
+            // Assert
+            expect(labelEl).not.toHaveAttribute("data-testid");
+            expect(descriptionEl).not.toHaveAttribute("data-testid");
+            expect(fieldEl).not.toHaveAttribute("data-testid");
+            expect(errorEl).not.toHaveAttribute("data-testid");
+        });
+
+        it("should persist original attributes on the field if it is not overridden", () => {
+            // Arrange
+            render(
+                <LabeledField
+                    field={
+                        <TextField
+                            value=""
+                            onChange={() => {}}
+                            name="name-example"
+                        />
+                    }
+                    label="Label"
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const fieldEl = screen.getByRole("textbox");
+
+            // Assert
+            expect(fieldEl).toHaveAttribute("name", "name-example");
+        });
+    });
+
+    describe("Accessibility", () => {
+        describe("Axe", () => {
+            it("should have no accessibility violations", async () => {
+                // Arrange
+                const {container} = render(
+                    <LabeledField
+                        field={<TextField value="" onChange={() => {}} />}
+                        label="Label"
+                        description="Description for the field"
+                        error="Error message"
+                    />,
+                    defaultOptions,
+                );
+                // Act
+
+                // Assert
+                await expect(container).toHaveNoA11yViolations();
+            });
+        });
+        describe("ARIA", () => {
+            it("should render a label tag with the for attribute set to the id of the field", () => {
+                // Arrange
+                const label = "Label";
+                render(
+                    <LabeledField
+                        field={<TextField value="" onChange={() => {}} />}
+                        label={label}
+                    />,
+                    defaultOptions,
+                );
+
+                // Act
+                const labelEl = screen.getByText(label);
+                const inputEl = screen.getByRole("textbox");
+
+                // Assert
+                expect(labelEl).toHaveAttribute("for", inputEl.id);
+            });
+
+            it("should set aria-describedby on the field to the id of the description", () => {
```

Comment:
Is there an example of this usage in Storybook? I saw the error example, but wasn't sure if this description variant was shown. It looks solid implementation-wise, though!

Reply:
Great question! All of the LabeledField stories had a description configured so we are able to inspect the `aria-describedby` relationship with the description! 

Your question made me realize that it would be helpful to have a story (and visual regression tests) for the different combinations of props since right now, all of the LabeledField stories show when both the label and description are present. I've added a new [Combinations story](https://5e1bf4b385e3fb0020b7073c-rnlzivmykl.chromatic.com/?path=/story/packages-labeledfield--combinations)!